### PR TITLE
Hot fix: update stats v4 endpoint and banner message when reverting from stats v4 to v3

### LINKED ISSUE #1489: Stats v4 reverts to Stats v3 if `wc-admin` plugin is v0.22 or higher
**Describe the bug**
Ref. p5T066-14o-p2. 

New Stats reverts to Old Stats when the `wc-admin` plugin is v0.22+. The API contains [breaking changes](https://href.li/?https://github.com/woocommerce/woocommerce-admin/pull/3204).

**To Reproduce**

1. On your test site, have the `wc-admin` plugin installed and updated to version 0.22. 
2. Visit your store on the mobile app. `We have reverted to the old stats as we couldn’t find the WooCommerce Admin plugin` error will appear.

The app logs will read:

```
Dashboard (Order Stats) — Error synchronizing order stats v4: Dotcom Invalid REST Route
Error synchronizing order stats: Dotcom Invalid REST Route
```

Making a request through Postman, the API error will read:

```
{
    "error": "rest_no_route",
    "message": "No route was found matching the URL and request method"
}
```

This is happening due to [a namespace change](https://href.li/?https://github.com/woocommerce/woocommerce-admin/pull/3204) to the API rest route.

**Screenshots**

**Expected behavior**
Check if the user is using the New Stats feature. Check if the user has an old version of `wc-admin`. If they are on an older version, display the message card (the one that invites users to try out the beta Stats) and say something like "Stats has been reverted to the non-beta version. To use the beta v4 stats please upgrade the wc-admin plugin to  version v0.22 or higher." Correct the API rest routes to use the new namespace. 

**Notes:** I'm not familiar with Stats implementation. Feel free to correct anything that's wrong with this bug report.

**Isolating the problem (mark completed items with an [x]):**
- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Mobile Environment**
Please include:
- Device: any
- iOS version: any 
- WooCommerce iOS version: 3.0
- wc-admin version: 0.22 and up

### PR DESCRIPTION
**Pull request by jaclync**
Fixes #1489 

## Changes
- Created a new Woo API version case `wcAnalytics = "wc-analytics"` for the new API namespace that replaces `mark4 = "wc/v4"` for stats in WC Admin plugin v0.22+
- Used the new WC analytics API namespace for the stats v4 endpoint
- In the banner where we show when the user was previously seeing stats v4 UI and now is not eligible for stats v4 (with the new namespace), updated the messaging to mention the minimum WC Admin plugin version (0.22) for new stats UI

### Test case 1: the store has WC Admin v0.22+

- In `wp-admin`, make sure the WC Admin plugin is v0.22+ and activated
- Launch the app from the PR branch --> should see the new stats, or the banner to view the new stats
- In `wp-admin`, deactivate the WC Admin plugin
- Back in the app, pull down to refresh or relaunch the app --> should see the stats reverted to v3 UI, and a banner suggesting updating the WC Admin plugin to v0.22+ as in the screenshot

### Test case 2: the store has WC Admin with a version lower than 0.22

- In `wp-admin`, make sure the WC Admin plugin is lower than v0.22 and activated (you can download an older WC Admin version from the [WC Admin GitHub releases page](https://github.com/woocommerce/woocommerce-admin/releases?after=v0.20.1))
- Launch the app from `develop` and opt-in the new stats if it's not currently shown
- Launch the app from the PR branch --> should see the stats reverted to v3 UI, and a banner suggesting updating the WC Admin plugin to v0.22+ as in the screenshot
- In `wp-admin`, update the WC Admin plugin to v0.22+ and make sure it's activated
- Launch the app from the PR branch --> should see the banner to view the new stats
- Tap "Try It Now" on the banner to view the new stats --> should see the new stats UI

## Example screenshots

When the user used to see stats v4 UI but isn't eligible for stats v4 anymore (either doesn't have WC Admin installed/activated or the WC Admin version is below 0.22), we show this banner:

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by jaclync** [comment]
@pmusolino @mindgraffiti could one of you take a look at this hotfix PR for release 3.1 when you get a chance today? More details in #1489. cc @JavonDavis 

As an overview for Paolo, we added a new stats UI for sites with WooCommerce Admin plugin activated. Here's a screenshot comparison btw v3 (old) and v4 (new) stats UI, I usually tell by the tab labels ("Days"/"Weeks"/"Months"/"Years" vs. "Today"/"This week"/"This month"/"This year"):

v3 stats UI | v4 stats UI
-- | --
Simulator Screen Shot - iPhone 11 - 2019-11-19 at 14 13 27 | Simulator Screen Shot - iPhone 11 - 2019-11-19 at 14 12 12

**Review by pmusolino — APPROVED** [review]
The code looks good to me! ✅ 

I do some tests, following your instructions.
Everything looks good, but the first screen (from develop branch, and on a new installation of the app) doesn't allow me to show the new stats. `Launch the app from develop and opt-in the new stats if it's not currently shown`. There is no opt-in banner. It's normal? 

| Develop branch, no opt-in message             |  After moving to the branch |
:-------------------------:|:-------------------------:
app develop no opt-in | branch

| Branch after opt-in             |  Branch with plugin deactivated |
:-------------------------:|:-------------------------:
branch opt-in | branch, plugin deactivated

**Review by mindgraffiti — COMMENTED** [review]
I can't get the new stats to take. I installed the old wc-admin (v.0.21) on my test site, then ran a fresh build on develop. Logged in and activated the new stats. Stopped the build, switched to the hotfix branch, and ran it. I saw this, which was good:



I went back to my website, updated the plugin to 0.22. Returned to the app and tried to pull-to-refresh. Old stats remained. I logged out of the app and logged back in. Old stats remained. I also don't have an option to opt-in to the new stats in Settings:

**Comment by jaclync** [comment]
> Launch the app from develop and opt-in the new stats if it's not currently shown. There is no opt-in banner. It's normal?

@pmusolino sorry I missed a condition in the test case - in order to see the new stats on `develop` the WC Admin plugin has to be lower than v0.22. I've updated the test cases (with a link to get the older versions of WC Admin plugin in the test case 2), could you try again? 😄

**Comment by jaclync** [comment]
> I went back to my website, updated the plugin to 0.22. Returned to the app and tried to pull-to-refresh. Old stats remained. I logged out of the app and logged back in. Old stats remained. I also don't have an option to opt-in to the new stats in Settings:

@mindgraffiti could you check on wp-admin that the WC Admin plugin is activated after the update (try refreshing the Plugins page)? If you don't see the option to opt-in to the new stats in Settings, that means we cannot get any data from the new stats endpoint so the new stats UI is not available. On my site, after I updated the plugin, somehow it got deactivated by default. After reactivating the plugin, I got to see the new stats UI again on the PR branch. See if this is the case for you, and if not, would it be possible to add me as a user to your test store?

**Comment by mindgraffiti** [comment]
Screen Shot 2019-11-20 at 8 03 23 AM

@jaclync, yep it deactivated from this error ☝️ and I reactivated it. I tried switching away from the store, coming back to it, using pull-to-refresh, and logging out and logging back in again. It doesn't usually take that long for the reactivated plugin to propagate the new API endpoints. Sending you an admin invite to my test site.

**Comment by mindgraffiti** [comment]
I dug a little deeper into this. It looks like the StatsVersionStateCoordinator doesn't take `wcAnalytics` into account, only stats .v4 and .v4. I also logged a few things and noticed that the v4 endpoint was getting requested, despite the the wcAnalytics namespace defined in the remote request.

**Comment by JavonDavis** [comment]
Just noting that even though https://github.com/woocommerce/woocommerce-android/pull/1575 is merged, I'm holding off on a new beta until this one is merged as well

**Comment by pmusolino** [comment]
> @pmusolino sorry I missed a condition in the test case - in order to see the new stats on develop the WC Admin plugin has to be lower than v0.22. I've updated the test cases (with a link to get the older versions of WC Admin plugin in the test case 2), could you try again? 😄

Now it worked 😉 ✅

**Comment by jaclync** [comment]
Hey @mindgraffiti I was able to see the new stats banner and then the UI in your store, though the Order stats all seem to be empty and consistent on wp-admin so it seems to be a separate issue.

Simulator Screen Shot - iPhone 8 - 2019-11-21 at 09 09 52

> I dug a little deeper into this. It looks like the StatsVersionStateCoordinator doesn't take wcAnalytics into account, only stats .v4 and .v4. (Could that be a factor?) I also logged a few things and noticed that the v4 endpoint was getting requested, despite the the wcAnalytics namespace defined in the remote request.

**Comment by jaclync** [comment]
tested on @mindgraffiti 's store and the test cases worked, merging now - thanks all for testing!

**Comment by jaclync** [comment]
@JavonDavis sorry for holding off the beta, it's now merged!

**Comment by mindgraffiti** [comment]
Thank you @jaclync! I appreciate it.

**Pull request merged**
